providers/collection_package.rb

```diff
@@ -1,94 +1,276 @@
+		Chef::Log.info "#{@new_resource} already exists - nothing to do."
+		if @current_resource.different
+			converge_by("Update #{@new_resource}") do
+				update_collection_package
+			end
+		else
+			Chef::Log.info "#{@new_resource} hasn't changed - nothing to do."
+		end
+	else
+		converge_by("Create #{@new_resource}") do
+			create_collection_package
+		end
+	end
 end
 
-def load_current_resource
-  @current_resource = Chef::Resource.resource_for_node(:opennms_collection_package, node).new(@new_resource.name)
+action :create_if_missing do
+	if @current_resource.exists
+		Chef::Log.info "#{@new_resource} already exists - nothing to do."
+	else
+		converge_by("Create #{@new_resource}") do
+			create_collection_package
+		end
+	end
+end
 
-  @current_resource.exists = true if package_exists?(@current_resource.name)
+action :delete do
+	if !@current_resource.exists
+		Chef::Log.info "#{@new_resource} doesn't exist - nothing to do."
+	else
+		converge_by("Delete #{@new_resource}") do
+			delete_collection_package
+		end
+	end
+end
+
+def load_current_resource
+	@current_resource = Chef::Resource.resource_for_node(:opennms_collection_package, node).new(@new_resource.name)
+	@current_resource.filter(@new_resource.filter)
+	@current_resource.specifics(@new_resource.specifics)
+	@current_resource.include_ranges(@new_resource.include_ranges)
+	@current_resource.exclude_ranges(@new_resource.exclude_ranges)
+	@current_resource.include_urls(@new_resource.include_urls)
+	@current_resource.store_by_if_alias(@new_resource.store_by_if_alias)
+	@current_resource.store_by_node_id(@new_resource.store_by_node_id)
+	@current_resource.if_alias_domain(@new_resource.if_alias_domain)
+	@current_resource.stor_flag_override(@new_resource.stor_flag_override)
+	@current_resource.if_alias_comment(@new_resource.if_alias_comment)
+	@current_resource.outage_calendars(@new_resource.outage_calendars)
+	
+	file = ::File.new("#{node['opennms']['conf']['home']}/etc/collectd-configuration.xml", 'r')
+	contents = file.read
+	file.close
+	
+	doc = REXML::Document.new(contents, respect_whitespace: :all)
+	doc.context[:attribute_quote] = :quote
+	package_el = matching_package_name(doc, @current_resource)
+	
+	if !package_el.nil?
+		@current_resource.exists = true
+	  if filter_change?(package_el, @current_resource.filter) || specifics_change?(package_el, @current_resource.specifics) || include_ranges_change?(package_el, @current_resource.include_ranges)
+			Chef::Log.debug("uei #{@current_resource.name} has changed.")
+			@current_resource.different = true
+	  end
+	end
 end
 
+
+
 private
 
+def matching_package_name(doc, current_resource)
+	definition = nil
+	doc.elements.each('/collectd-configuration/package') do |package_name|
+		next unless package_name.attributes['name'].to_s == current_resource.to_s
+		definition = package_name
+		break
+	end
+	definition
+end
+
+def filter_change?(package_el, current_filter)
+	filter_el = package_el.elements['filter'].text.to_s
```

Comment:
The `text` method only gets the first text element, and there could be more. What I typically do is use `texts.join("\n")`.